### PR TITLE
Log all peer dependency warnings from yarn and npm

### PR DESCRIPTION
**Pull request by feelepxyz**

### CODE REVIEW

**Comment by greysteil** [comment]
Nice! Want me to find some test files and get that set up?

**Comment by feelepxyz** [comment]
@greysteil haven't added the tests yet, sorting some stuff out with a friend so will do tomorrow.

Found found this interesting case:

```
{
    "name": "test",
    "version": "1.0.0",
    "private": true,
    "dependencies": {
        "apollo-client": "^0.8.0",
        "react": "^15.0.0",
        "react-dom": "^15.0.0",
        "react-apollo": "^0.9.0"
    }
}
```

If you update react to 16 you get two peer dep warnings:

```
" > react-dom@15.6.2" has incorrect peer dependency "react@^15.6.2".
" > react-apollo@0.9.0" has incorrect peer dependency "react@0.14.x || 15.* || ^15.0.0"
```

But then if you update `react-dom` and `react-apollo` to the latest you also need to update `apollo-client` to `^2.3.8` and install `graphql`.. 🤠 

If you go even further back with react-apollo it also has a peer dep on `redux`... 🤯

**Comment by feelepxyz** [comment]
@greysteil go for it if you're looking at this otherwise I'll sort it out tomorrow!

**Comment by greysteil** [comment]
@feelepxyz nice one!

I've added a couple of tests but they're currently failing. If you print the error output for both of them you'll see that only one peer dependency warning (the `react-apollo` one) is coming through. The `react-dom` one is getting missed for both Yarn and npm.

**Comment by greysteil** [comment]
Also of note: peer dependencies *do* apply to sub-dependencies. Let's not worry about them for now(!).

File: spec/dependabot/update_checkers/java_script/npm_and_yarn/version_resolver_spec.rb [hunk 1]
```diff
@@ -535,6 +555,26 @@
...
+              requirements: [{
+                file: "package.json",
+                requirement: "0.14.2",
+                groups: ["dependencies"],
+                source: nil
+              }]
+            )
+          end
+
+          pending { is_expected.to eq(Gem::Version.new("0.14.9")) }
```
Comment (greysteil) [inline spec/dependabot/update_checkers/java_script/npm_and_yarn/version_resolver_spec.rb]: @feelepxyz - weirdly, it doesn't look like the peer dependency checker is finding *any* errors for this case. It should be - updating from `0.14.2` to `16.3.1` generates lots of them when done locally. Any ideas?

**Comment by feelepxyz** [comment]
@greysteil think this works as expected now, spent ages looking for packages (with few deps) that could be installed with a range that wouldn't give you lax peer dependency requirements. Most packages will relax their peer dependency requirements to allow new React major versions in a patch/minor release!

**Comment by greysteil** [comment]
Nice!

**Pull request merged**